tensorboard/webapp/metrics/store/metrics_store_internal_utils_test.ts

```diff
@@ -433,4 +439,61 @@ describe('metrics store utils', () => {
       expect(result.cardMetadataMap).toEqual(originals.cardMetadataMap);
     });
   });
+
+  describe('canCreateNewPins', () => {
+    it('returns true when pins are under the limit', () => {
+      TEST_ONLY.util.MAX_PIN_COUNT = 3;
+      const state = buildMetricsState({
+        pinnedCardToOriginal: new Map([['pinnedCard1', 'card1']]),
+        unresolvedImportedPinnedCards: [
+          {plugin: PluginType.SCALARS, tag: 'loss'},
+        ],
+      });
+
+      expect(canCreateNewPins(state)).toBe(true);
+    });
+
+    it('returns false when resolved pins reaches the limit', () => {
+      TEST_ONLY.util.MAX_PIN_COUNT = 3;
+      const state = buildMetricsState({
+        pinnedCardToOriginal: new Map([
+          ['pinnedCard1', 'card1'],
+          ['pinnedCard2', 'card2'],
+          ['pinnedCard3', 'card3'],
+        ]),
+        unresolvedImportedPinnedCards: [],
+      });
+
+      expect(canCreateNewPins(state)).toBe(false);
+    });
+
+    it('returns false when unresolved pins reaches the limit', () => {
+      TEST_ONLY.util.MAX_PIN_COUNT = 3;
```

Comment:
umm, you are mutating the global state and it won't restore itself. Maybe use `spyOnProperty` which should restore the original value.

Reply:
Ah, added an `afterEach` to restore the original value.

It seems `spyOnProperty` does not work for object literal properties, only getter properties or ones created with Object.defineProperty.